lib/ask/activity_log.ex

```diff
@@ -103,21 +103,28 @@ defmodule Ask.ActivityLog do
   defp typeof(%Questionnaire{}), do: "questionnaire"
   defp typeof(%Folder{}), do: "folder"
 
+  defp remote_ip(nil), do: "0.0.0.0"
+
+  defp remote_ip(%{remote_ip: remote_ip, req_headers: headers}) do
+    case Enum.find(headers, fn {key, _value} -> key == "x-forwarded-for" end) do
+      {"x-forwarded-for", forwarded_ip} -> forwarded_ip
+      nil -> to_string(:inet_parse.ntoa(remote_ip))
+    end
+  end
+
   defp create(action, project, conn, entity, metadata) do
-    {user_id, remote_ip} =
-      case conn do
-        nil ->
-          {nil, "0.0.0.0"}
-
-        conn ->
-          remote_ip = to_string(:inet_parse.ntoa(conn.remote_ip))
-
-          case current_user(conn) do
-            nil -> {nil, remote_ip}
-            user -> {user.id, remote_ip}
-          end
+    user_id =
+      if conn do
+        case current_user(conn) do
+          nil -> nil
+          user -> user.id
+        end
+      else
+        nil
```

Comment:
wouldn't it make more sense to add a guard for the `current_user` that handles the a `nil` parameter?
```elixir
  def current_user(nil) do
    nil
  end
```

and here just do
```elixir
user_id = case current_user(conn) do
           nil -> nil
           user -> user.id
          end
```

Reply:
I'm not 100% convinced that we should _not_ crash if we get a `nil` `conn`, but I've implemented the change anyways 👍